.github/workflows/codecov_main.yaml

```diff
@@ -4,53 +4,44 @@ on:
     branches:
       - main
 
+defaults:
+  run:
+    working-directory: ./backend
+
 jobs:
   test-backend:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11"]
+        python-version: ["3.10", "3.11", "3.12"]
         os: [ubuntu-latest]
         #os: [ ubuntu-latest, windows-latest ]
 
     steps:
       - uses: actions/checkout@v4
+
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
 
-      #----------------------------------------------
-      #          install & configure poetry
-      #----------------------------------------------
-      - name: Install Poetry
-        uses: snok/install-poetry@v1
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
 
-      #----------------------------------------------
-      #    install your root project, if required
-      #----------------------------------------------
```

Comment:
i would implement a check to verify lock with toml just to make sure both are in sync 
```
      - name: Verify lockfile is up-to-date
        run: |
          uv lock --check
```

Reply:
i think it is safest to do it before any `uv sync`?
as `uv sync` is not checking if lock and toml are out of sync
and it does not veryify if the pyproject.toml has changed since the lock was created.
this is basically just a quaility check if someone forgets to run `uv lock `